src/components/ExportNotesForm.jsx

```diff
@@ -0,0 +1,65 @@
+import React, { useState } from 'react'
+
+const ExportNotesForm = ({
+  show,
+  onSubmit,
+  onCancel
+}) => {
+  const [ filename, setFilename ] = useState('mynotes')
+  const [ format, setFormat ] = useState('json')
+
+  const onFilenameChange = e => {
+    let value = e.target.value || ''
+    value = value
+      .replace(/[/?<>\\:*|"]/g, '')
+      // eslint-disable-next-line no-control-regex
+      .replace(/[\x00-\x1f\x80-\x9f]/g, '')
+      .replace(/^\.+$/, '')
+      .replace(/\s/g, '_')
+      .substr(0, 255)
+    setFilename(value)
+  }
+
+  const onFormatChange = e => {
+    setFormat(e.target.value)
+  }
+
+  const submitForm = e => {
+    e && e.preventDefault()
+    e && e.stopPropagation()
+    typeof onSubmit === 'function' && onSubmit({
+      filename: `${filename}.${format}`,
+      format
+    })
+  }
+
+  const cancelForm = e => {
```

Comment:
not a fan of single letter variable names, but just my opinion

Reply:
I'm not either, but I also don't like using "event" since it shadows the global "event" nor am I an advocate for using the horribly abbreviated "evt" "evnt" variants.

e as the only argument is typically what is accepted.